termux-shared/src/main/java/com/termux/shared/file/FileUtils.java

```diff
@@ -0,0 +1,1627 @@
+        return copyOrMoveFile(context, label, srcFilePath, destFilePath,
+            true, ignoreNonExistentSrcFile, FileTypes.FILE_TYPE_NORMAL_FLAGS,
+            true, true);
+    }
+    
+    /**
+     * Copy or move a file from {@code sourceFilePath} to {@code destFilePath}.
+     *
+     * The {@code sourceFilePath} and {@code destFilePath} must be the canonical path to the source
+     * and destination since symlinks will not be followed.
+     *
+     * If the {@code sourceFilePath} or {@code destFilePath} is a canonical path to a directory,
+     * then any symlink files found under the directory will be deleted, but not their targets when
+     * deleting source after move and deleting destination before copy/move.
+     *
+     * @param context The {@link Context} to get error string.
+     * @param label The optional label for file to copy or move. This can optionally be {@code null}.
+     * @param srcFilePath The {@code source path} for file to copy or move.
+     * @param destFilePath The {@code destination path} for file to copy or move.
+     * @param moveFile The {@code boolean} that decides if source file needs to be copied or moved.
+     *                 If set to {@code true}, then source file will be moved, otherwise it will be
+     *                 copied.
+     * @param ignoreNonExistentSrcFile The {@code boolean} that decides if it should be considered an
+     *                              error if source file to copied or moved doesn't exist.
+     * @param allowedFileTypeFlags The flags that are matched against the source file's {@link FileType}
+     *                             to see if it should be copied/moved or not. This is a safety measure
+     *                             to prevent accidental copy/move/delete of the wrong type of file,
+     *                             like a directory instead of a regular file. You can pass
+     *                             {@link FileTypes#FILE_TYPE_ANY_FLAGS} to allow copy/move of any file type.
+     * @param overwrite The {@code boolean} that decides if destination file should be overwritten if
+     *                  it already exists. If set to {@code true}, then destination file will be
+     *                  deleted before source is copied or moved.
+     * @param overwriteOnlyIfDestSameFileTypeAsSrc The {@code boolean} that decides if overwrite should
+     *                                         only be done if destination file is also the same file
+     *                                          type as the source file.
+     * @return Returns the {@code errmsg} if copy or move was not successful, otherwise {@code null}.
+     */
+    public static String copyOrMoveFile(@NonNull final Context context, String label, final String srcFilePath, final String destFilePath,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 31. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.